app/src/main/java/com/github/pockethub/android/ui/notification/NotificationListFragment.java

```diff
@@ -0,0 +1,181 @@
+import com.github.pockethub.android.R;
+import com.github.pockethub.android.core.issue.IssueUriMatcher;
+import com.github.pockethub.android.ui.DialogFragment;
+import com.github.pockethub.android.ui.issue.IssuesViewActivity;
+import com.github.pockethub.android.util.ToastUtils;
+import com.meisolsson.githubsdk.core.ServiceGenerator;
+import com.meisolsson.githubsdk.model.Issue;
+import com.meisolsson.githubsdk.model.NotificationThread;
+import com.meisolsson.githubsdk.model.Page;
+import com.meisolsson.githubsdk.model.Repository;
+import com.meisolsson.githubsdk.model.request.NotificationReadRequest;
+import com.meisolsson.githubsdk.service.activity.NotificationService;
+
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NotificationListFragment extends DialogFragment
+        implements NotificationReadListener, AdapterView.OnItemClickListener {
+
+    /**
+     * Filters for the request to GitHub.
+     */
+    private Map<String, Object> filters = new HashMap<>();
+
+    private NotificationListAdapter adapter;
+
+    private ListView list;
+    private TextView emptyText;
+    private ProgressBar progressBar;
+    private SwipeRefreshLayout refreshLayout;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle args = getArguments();
+        if (args.containsKey("filter")) {
+            filters.put(args.getString("filter"), true);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_item_list, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        refreshLayout = (SwipeRefreshLayout) view.findViewById(R.id.swipe_item);
+        refreshLayout.setOnRefreshListener(this::refreshNotifications);
+
+        emptyText = (TextView) view.findViewById(android.R.id.empty);
+        emptyText.setText(R.string.no_notifications);
+
+        progressBar = (ProgressBar) view.findViewById(R.id.pb_loading);
+        list = (ListView) view.findViewById(android.R.id.list);
+        list.setVisibility(View.VISIBLE);
+        list.setOnItemClickListener(this);
+
+        adapter = new NotificationListAdapter(getActivity().getLayoutInflater(), this);
+
+        list.setAdapter(adapter);
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        refreshNotifications();
+    }
+
+    private void refreshNotifications() {
```

Comment:
I think that this method would be a bit more readable and concise if it was written as follow:

```
private void refreshNotifications() {
    list.setVisibility(View.GONE);
    progressBar.setVisibility(View.VISIBLE);
    getPageAndNext(1)
            .flatMap(page -> Observable.fromIterable(page.items()))
            .toList()
            .subscribe(threads -> {
                adapter.setItems(threads);
                progressBar.setVisibility(View.GONE);
                list.setVisibility(View.VISIBLE);
                emptyText.setVisibility(threads.size() == 0 ? View.VISIBLE : View.GONE);
                refreshLayout.setRefreshing(false);
            }, e -> ToastUtils.show(getActivity(), R.string.error_notifications_load));
}
```